code/game/machinery/kitchen/microwave.dm

```diff
@@ -167,86 +183,98 @@
+			to_chat(user, "<span class='warning'>This [src] is full of ingredients, you can't fit any more!</span>")
+			return 1
+		if(istype(O, /obj/item/stack))
+			var/obj/item/stack/S = O
+			if(S.get_amount() > 1)
+				new O.type (src)
+				S.use(1)
+				user.visible_message( \
+					"<span class='notice'>\The [user] has added one of [O] to \the [src].</span>", \
+					"<span class='notice'>You add one of [O] to \the [src].</span>")
+				SSvueui.check_uis_for_change(src)
+				return
+			else
+				add_item(O, user)
+		else
+			add_item(O, user)
+			return
+	SSvueui.check_uis_for_change(src)
 	..()
-	src.updateUsrDialog()
+
+/obj/machinery/microwave/AltClick()
+	if(!operating)
+		cook()
 
 /obj/machinery/microwave/attack_ai(mob/user as mob)
 	if(istype(user, /mob/living/silicon/robot) && Adjacent(user))
 		attack_hand(user)
 
 /obj/machinery/microwave/attack_hand(mob/user as mob)
 	user.set_machine(src)
-	interact(user)
+	if(broken > 0)
+		to_chat(user, "<span class='warning'>\The [name] is broken! You'll need to fix it before using it.</span>")
+	else if(dirty >= 100)
+		to_chat(user, "<span class='warning'>\The [name] is dirty! You'll need to clean it before using it.</span>")
+	else
+		ui_interact(user)
+
+/obj/machinery/microwave/examine(var/mob/user)
+	..()
+	if(broken > 0)
+		to_chat(user, "It's broken!")
+	else if(dirty >= 100)
+		to_chat(user, "The insides are completely filthy!")
+	else if(dirty > 75)
+		to_chat(user, "It's covered in stains.")
+	else if(dirty > 50)
+		to_chat(user, "It's pretty messy.")
+	else if(dirty > 25)
+		to_chat(user, "It's a bit dirty.")
 
 /*******************
 *   Microwave Menu
 ********************/
 
-/obj/machinery/microwave/interact(mob/user as mob) // The microwave Menu
-	var/dat = ""
-	if(src.broken > 0)
-		dat = {"<TT>Bzzzzttttt</TT>"}
-	else if(src.operating)
-		dat = {"<TT>Microwaving in progress!<BR>Please wait...!</TT>"}
-	else if(src.dirty==100)
-		dat = {"<TT>This microwave is dirty!<BR>Please clean it before use!</TT>"}
-	else
-		var/list/items_counts = new
-		var/list/items_measures = new
-		var/list/items_measures_p = new
+VUEUI_MONITOR_VARS(/obj/machinery/microwave, microwavemonitor)
+	watch_var("operating", "on", CALLBACK(null, .proc/transform_to_boolean, FALSE))
+	watch_var("failed", "failed")
+	watch_var("cook_time", "cook_time")
+	watch_var("start_time", "start_time")
+
+/obj/machinery/microwave/vueui_data_change(var/list/data, var/mob/user, var/datum/vueui/ui)
+	var/monitordata = ..()
+	if(monitordata)
+		. = data = monitordata
+
+	// init data objects if they aren't already
+	LAZYINITLIST(data["cookingobjs"])
+	LAZYINITLIST(data["cookingreas"])
+
+	// if BYOND lists are smaller than UI, then something (or everything) was removed - wipe the list
+	if(LAZYLEN(contents) < LAZYLEN(data["cookingobjs"]))
+		VUEUI_SET_CHECK(data["cookingobjs"], list(), ., data)
+	if(LAZYLEN(reagents.reagent_list) < LAZYLEN(data["cookingreas"]))
+		VUEUI_SET_CHECK(data["cookingreas"], list(), ., data)

```

Comment:
Maybe you should directly alter UI data when removing ingredients? 
by doing something like:
```dm
for(var/i in SSvueui.get_open_uis(src))
    var/datum/vueui/ui = i
    ui.data["cookingobjs"] -= "[rmeovedobj]"

Reply:
Would that be significantly better? You warned me against altering UI data directly before.

Reply:
Well, might not be a huge difference, but if it happens a lot it might become an issue. Might be fine tho. Also directly editing UI object data is not recommended, but if its only way to get things to properly update more complex structures then it's okay. Like I am sure I am editing UI data inside record computers.